sway-lib-std/src/u128.sw

```diff
@@ -0,0 +1,310 @@
+library u128;
+
+use core::num::*;
+use ::assert::assert;
+use ::result::Result;
+
+/// The 128-bit unsigned integer type.
+/// Represented as two 64-bit components: `(upper, lower)`, where `value = (upper << 64) + lower`.
+pub struct U128 {
+    upper: u64,
+    lower: u64,
+}
+
+pub trait From {
+    /// Function for creating U128 from its u64 components.
+    pub fn from(h: u64, l: u64) -> Self;
+} {
+    fn into(v: U128) -> (u64, u64) {
+        (v.upper, v.lower)
+    }
+}
+
+impl From for U128 {
+    pub fn from(h: u64, l: u64) -> U128 {
+        U128 {
+            upper: h,
+            lower: l,
+        }
+    }
+}
+
+impl core::ops::Eq for U128 {
+    pub fn eq(self, other: Self) -> bool {
+        self.lower == other.lower && self.upper == other.upper
+    }
+}
+
+impl core::ops::Ord for U128 {
+    pub fn gt(self, other: Self) -> bool {
+        self.upper > other.upper || self.upper == other.upper && self.lower > other.lower
+    }
+
+    pub fn lt(self, other: Self) -> bool {
+        self.upper < other.upper || self.upper == other.upper && self.lower < other.lower
+    }
+}
+
+// TODO this doesn't work?
+// impl core::ops::OrdEq for U128 {
+// }
+
+fn disable_overflow() {
+    asm(r1) {
+        movi r1 i3;
+        flag r1;
+    }
+}
+
+fn enable_overflow() {
+    asm(r1) {
+        movi r1 i0;
+        flag r1;
+    }
+}
```

Comment:
hmm.  My understanding was that the `flag` op had to be used in the same asm block as the op you're trying to allow overflow for, otherwise, the bit is unset after the block returns. 
Is this working?

Reply:
Also, perhaps rather than `movi r1 i3;` here you could initialize a second register with the value as a named variable (or even better, as a `const`) to:
   a.) help document the function better, and
   b.) to follow the general guideline of "if it can be done outside of asm, do it outside of asm".
   
   ie:
   ```
const WRAPPING_FLAG_VALUE = 3;
asm(r1: WRAPPING_FLAG_VALUE) {
    flag r1;
}
```

Reply:
And finally, it seems that the flag values for `enable` and `disable` are reversed.
An unset flag (ie: `0`) should be the default state in which panic occurs on overflow.
Setting to a non-zero value should enable overflow.

Reply:
> Also, perhaps rather than `movi r1 i3;` here you could initialize a second register with the value as a named variable (or even better, as a `const`) to: a.) help document the function better, and b.) to follow the general guideline of "if it can be done outside of asm, do it outside of asm".
> 
> ie:
> 
> ```
> const WRAPPING_FLAG_VALUE = 3;
> asm(r1: WRAPPING_FLAG_VALUE) {
>  flag r1;
> }
> ```

What about
```
asm(WRAPPING_FLAG_VALUE: 3) {
    flag WRAPPING_FLAG_VALUE;
}
```

Reply:
I think a combination of a descriptively-named register and a const value would be best, and in line with `tx.sw` and other modules where we use named const values to initialize registers with.

Reply:
>Do these belong somewhere more general as pub definitions? Like num or somewhere?

We had talked about a stdlib module for wrapping VM opcodes, this would probably fit the bill. Will look for the open issue.

Reply:
@otrho given that no other part of the stdlib needs these functions, can we defer it to a future PR?